src/main/java/com/zufar/onlinestore/product/endpoint/ProductsEndpoint.java

```diff
@@ -25,13 +28,15 @@
 @Validated
 @RequestMapping(value = "/api/products")
 public class ProductsEndpoint {
+
     private final ProductInfoRepository productInfoRepository;
     private final ProductInfoDtoConverter productInfoDtoConverter;
+    private final ProductInfoService productInfoService;
 
     @GetMapping("/{id}")
     public ResponseEntity<ProductInfoDto> getProductInfoById(@PathVariable("id") final String id) {
         log.info("Received request to get the ProductInfo with id - {}.", id);
-        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(Integer.parseInt(id));
+        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(UUID.fromString(id));
```

Comment:
create `getProductById` method in `ProductInfoService` and `ProductInfoServiceImpl` and use this class here like `productInfoService.getProductById(productId);`